components/Buttons/tests/unit/ButtonsThemingTest.swift

```diff
@@ -84,4 +84,62 @@ class ButtonsThemingTest: XCTestCase {
     XCTAssertEqual(buttonShape.bottomRightCorner, shapeScheme.smallComponentShape.bottomRightCorner)
     XCTAssertEqual(buttonShape.bottomLeftCorner, shapeScheme.smallComponentShape.bottomLeftCorner)
   }
+  
+  func testFloatingButtonSecondaryThemeWithNoCustomThemes() {
+    // Given
+    let button = MDCFloatingButton()
+    let scheme = MDCContainerScheme()
+    let colorScheme = MDCSemanticColorScheme(defaults: .material201804)
+    let typographyScheme = MDCTypographyScheme(defaults: .material201804)
+
+    // When
+    button.applySecondaryTheme(withScheme: scheme)
+
+    // Then
+    XCTAssertEqual(button.backgroundColor(for: .normal), colorScheme.secondaryColor)
+    XCTAssertEqual(button.imageTintColor(for: .normal), colorScheme.onSecondaryColor)
+
+    XCTAssertEqual(button.titleFont(for: .normal), typographyScheme.button)
+
+    let buttonShape = button.shapeGenerator as! MDCRectangleShapeGenerator
+    XCTAssertNotNil(buttonShape)
+    let corner = MDCCornerTreatment.corner(withRadius: 0.5)
+    corner?.valueType = .percentage
+    XCTAssertEqual(buttonShape.topLeftCorner, corner)
+    XCTAssertEqual(buttonShape.topRightCorner, corner)
+    XCTAssertEqual(buttonShape.bottomRightCorner, corner)
+    XCTAssertEqual(buttonShape.bottomLeftCorner, corner)
+  }
+
+  func testFloatingButtonSecondaryThemeWithCustomColorTheme() {
+    // Given
+    let button = MDCFloatingButton()
+    let scheme = MDCContainerScheme()
+    let customColorScheme = MDCSemanticColorScheme()
+    customColorScheme.secondaryColor = .purple
+    customColorScheme.onSecondaryColor = .purple
```

Comment:
Please make these colors distinct so `secondaryColor` and `onSecondaryColor` can be distinguished.  Please also customize the rest of the scheme colors.  If it's easier, you can do something like:

```
customColorScheme.primaryColor = [UIColor colorWithWhite:0 alpha:(CGFloat)0.05];
customColorScheme.secondary = [UIColor colorWithWhite:0 alpha:(CGFloat)0.1];
customColorScheme.surfaceColor = [UIColor colorWithWhite:0 alpha:(CGFloat)0.15];
customColorScheme.backgroundColor = [UIColor colorWithWhite:0 alpha:(CGFloat)0.2];
...
```